### PR TITLE
assignment4 1bbartels

### PR DESCRIPTION
**Pull request by 1bbartels**

### CODE REVIEW

File: Logger/IEntity.cs [hunk 1]
```diff
@@ -1,7 +1,6 @@
 ﻿namespace Logger;
 public interface IEntity
 {
-
-        // Place members here.
-
+    Guid Id { get; init; }
+    string? Name { get; }
```
Comment (alowe4) [inline Logger/IEntity.cs:5]: The question mark on string feels weird, why would there ever be an entity without a name?

**Review by alowe4 — COMMENTED** [review]
over all look great!
______________________
Tasks
    Define a full name record (first, last, middle) handling optional and null appropriately. ✔
        Provide a comment on the full name record on why you selected to define a value or a reference type and ✔
        Provide a comment on the full name record on why or why not the type is immutable. ✔
    Define an IEntity interface: ✔
        Add an Id property of type Guid that is an init-only setter. ✔
        Add a Name property that is string. ✔
    Define an abstract base class that implements IEntity - appropriately choosing to implement the interface explicitly or implicitly. ✔
        Do not implement the Name property in this abstract class. ✔
        Do force any derived classes to provide an implementation for Name. ✔
    Define book, student, and employee records - all with entity capabilities. ✔
        Choose a thoughtful approach for the value of Name for each type of entity. In all cases, there should be no backing field for Name (not even one generated by an automatically implemented property) - consider a calculated property. ✔
        Provide a comment on each interface member in each entity explaining why you implemented it implicitly or explicitly. ✔
        You should consider the relationship between Student and Employee and refactor the common code shared between them. ✔
        Given the properties on the full name record, you should consider which entities it makes sense to use it. ✔
        Test that the equality behavior on these entities behaves as expected. ✔
        Provide a comment on each interface method explaining why you implemented it implicitly or explicitly.✔
    Unit test the Storage class using various entities to ensure that there are no bugs.✔
    Code review (using TODO: comments if there are any issues - which there may not be) all logger-related classes.)
        Pay special attention to the CreateLogger methods throughout. ✔

Fundamentals
    Be sure you enable:
        Enable Nullability for all projects ✔
        Set LangVersion and the TargetFramework to the latest released versions available (preview versions optional) ✔
        Ensure that you turn on code analysis for all projects(EnableNETAnalyzers) ✔
    Ensure there are no errors or warnings (including code analysis warnings) ✔
    All of the above should be unit tested ✔
    Choose simplicity over complexity ✔

Extra Credit
    Refactor your project files to put common properties and/or items into a single file. ❌
    Convert ILogger.CreateLogger to a generic method. ❌
        Returns the logger classes type that is to be created. ❌
        Has a type parameter for the configuration type ❌

**Comment by crawfordkid2** [comment]
ill be reviewing this

**Review by crawfordkid2 — COMMENTED** [review]
Define a full name record (first, last, middle) handling optional and null appropriately. ✔
Provide a comment on the full name record on why you selected to define a value or a reference type and ✔
Provide a comment on the full name record on why or why not the type is immutable. ✔
Define an IEntity interface: ✔
Add an Id property of type Guid that is an init-only setter. ✔
Add a Name property that is string. ✔
Define an abstract base class that implements IEntity - appropriately choosing to implement the interface explicitly or implicitly. ✔
Do not implement the Name property in this abstract class. ✔
Do force any derived classes to provide an implementation for Name. ✔
Define book, student, and employee records - all with entity capabilities. ✔
Choose a thoughtful approach for the value of Name for each type of entity. In all cases, there should be no backing field for Name (not even one generated by an automatically implemented property) - consider a calculated property. ✔
Provide a comment on each interface member in each entity explaining why you implemented it implicitly or explicitly. ✔
You should consider the relationship between Student and Employee and refactor the common code shared between them. ✔
Given the properties on the full name record, you should consider which entities it makes sense to use it. ✔
Test that the equality behavior on these entities behaves as expected. ✔
Provide a comment on each interface method explaining why you implemented it implicitly or explicitly.
Unit test the Storage class using various entities to ensure that there are no bugs.
Code review (using TODO: comments if there are any issues - which there may not be) all logger-related classes.)
Pay special attention to the CreateLogger methods throughout.
Fundamentals
Be sure you enable:
Enable Nullability for all projects ✔
Set LangVersion and the TargetFramework to the latest released versions available (preview versions optional) ✔
Ensure that you turn on code analysis for all projects(EnableNETAnalyzers) ✔
Ensure there are no errors or warnings (including code analysis warnings) ❌
All of the above should be unit tested ✔
Choose simplicity over complexity ✔

NOTES:
why not put all student, employee, book, and person into the same file. These classes dont have much in them, just for simplicity, you can have all the class declarations in the same file if the purpose is similar. they are all doing basically the same thing.
- add treat warnings as errors and codeanalysistreatwarnings as errors

File: Logger/AbstractBaseEntity.cs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Logger
+{
+    public abstract class AbstractBaseEntity : IEntity
```
Comment (mmwoodfo) [inline Logger/AbstractBaseEntity.cs:9]:
```suggestion
    public abstract class BaseEntity : IEntity
```
Typically you wouldn't put `abstract` in the class name

**Review by mmwoodfo — COMMENTED** [review]
Good job overall! Looks like there were a few comments left by other students that also had some good tips.

**Pull request closed without merging**